.github/scripts/report_git_status.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+CHANGES=$(git status --porcelain)
+echo "$CHANGES"
+git diff
+[ -z "$CHANGES" ]
```

Comment:
i am not sure what this does. does it control the return code?

Reply:
you mean the last line specifically? it makes the workflow fail if the Git status is nonempty

Reply:
yup. exactly what i meant for return code :-)